src/test/java/de/unistuttgart/overworldbackend/WorldControllerTest.java

```diff
@@ -0,0 +1,171 @@
+package de.unistuttgart.overworldbackend;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.unistuttgart.overworldbackend.data.Lecture;
+import de.unistuttgart.overworldbackend.data.World;
+import de.unistuttgart.overworldbackend.data.WorldDTO;
+import de.unistuttgart.overworldbackend.data.mapper.WorldMapper;
+import de.unistuttgart.overworldbackend.repositories.LectureRepository;
+import de.unistuttgart.overworldbackend.repositories.WorldRepository;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@AutoConfigureMockMvc
+@SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class WorldControllerTest {
+
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private LectureRepository lectureRepository;
+
+  @Autowired
+  private WorldMapper worldMapper;
+
+  private final String API_URL = "/api/v1/overworld";
+  private String fullURL;
+  private ObjectMapper objectMapper;
+
+  private Lecture initialLecture;
+  private World initialWorld;
+  private WorldDTO initialWorldDTO;
+
+  @BeforeEach
+  public void createBasicData() {
+    lectureRepository.deleteAll();
+
+    final World world = new World();
+    world.setStaticName("Winter Wonderland");
+    world.setTopicName("UML Winter");
+    world.setActive(true);
+    world.setMinigameTasks(Set.of());
+    world.setNpcs(Set.of());
+    world.setDungeons(Arrays.asList());
+
+    final Lecture lecture = new Lecture("PSE", "Basic lecture of computer science students", Arrays.asList(world));
+    initialLecture = lectureRepository.save(lecture);
+    initialWorld = initialLecture.getWorlds().stream().findFirst().get();
+    initialWorldDTO = worldMapper.worldToWorldDTO(initialWorld);
+
+    assertNotNull(initialWorld.getId());
+    assertNotNull(initialWorldDTO.getId());
+
+    fullURL = "/lectures/" + initialLecture.getId() + "/worlds";
```

Comment:
Isn't there
```suggestion
    fullURL = API_URL + "/lectures/" + initialLecture.getId() + "/worlds";
```
missing?

Reply:
since we defined `/api/v1/overworld` in here `server.servlet.context-path=/api/v1/overworld` in _application.propterties_ the tests do not need the prefix `/api/v1/overworld`. So API_URL is an unused variable at the moment. I will remove it